keps/sig-windows/2578-windows-conformance/README.md

```diff
@@ -0,0 +1,437 @@
+
+#### Story 2
+
+As a Kubernetes vendor I want to evaluate the completeness of my Kubernetes support matrix in context of Windows supported features in the Kubernetes API.
+
+#### Story 3
+
+As a Kubernetes developer I want to immediately know whether or not the features I want to add will affect the Windows support matrix. If there is a confirmed or potential impact I want to be able to rapidly convey this in a canonical way to the broader community.
+
+#### Story 4
+
+As an IT administrator, I want to know whether my current version of Kubernetes supports ActiveDirectory or other Windows features.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Specification of Windows Operational Readiness (Kubernetes 1.24)
+
+- In order to qualify as a Windows test, the pods that demonstrate functionality for a given attribute of a cluster must be running on a `pause` image. 
+
+Additionally, this image must be scheduled to an operating system that meets the following requirements:
+
+- Windows Build Versions:
+  - Windows Server 2019 (LTSC)
+  - Windows Server 2022 (LTSC)
+
+- Installed KBs:
+  - Kubernetes 1.21 and forward
+
+- All tests in the Core Windows Operational Readiness suite depend on `pause` as well as `agnhost`, and no other images.  This is because there are no requirements (thus far) in this specification which cannot be tested via the `pause` image, and thus it simplifies the types of pods we test on.  In the future we may changes this requirement if new workloads are important for operational readiness specification (i.e. a GPU intensive app or a CNI for host process verification).
+- All tests should use the `Pod.OS` specification field as the *only* required scheduling directive for pods, once this field is made GA.  Up until this time, the way tests schedule Windows pods can use RuntimeClasses, Labels, or taints in any way which "works" to implement the other requirements.
+  - until this field is fully available in all tests, the use of  `tolerations` and `labels` should be used for all tests to solidify the scheduling aspects of windows containers. 
+ 
+#### Basic Networking
+
+
+- Ability to access Linux container IPs *by service IP* from Windows containers
+- Ability to access Windows container IPs *by service IP* from Linux containers
+- Ability to access Linux container IPs *by NodePort IP* from Windows containers
+- Ability to access Windows container IPs *by NodePort IP* from Linux containers
+- Ability to access Linux container IPs *by pod IP* from Windows containers
+- Ability to access Windows container IPs *by pod IP* from Linux containers
+- Ability to schedule multiple containers, with distinct IP addresses, on the same node.
+- Ability to delete and recreate services in such a way that load balancing rules for pods are recovered by whatever chosen service proxy is being utilized.
```

Comment:
Is there a reason we are not specifying kube-proxy as the service proxy here?

Reply:
Additionally, is the vagueness here surrounding the service proxy choice intentional? I believe that we should initially standardize on kube-proxy for Windows as it is the only viable **and** currently available service proxy for Windows that I am aware of. Leaving it as-is could result in anyone falling into our user stories above to interpret this as _any_ service proxy drop in can pass our Windows Operational Readiness conformance tests.

Reply:
@jayunit100 @jsturtevant @knabben any thoughts?

Reply:
I think the most used proxy is winkernel (used by azurecni, calico, flannel, and win-brdige) but I believe Antrea uses the userspace proxy.  

If someone were to develop a proxy that met all these requirements and passed the test is there a reason to block them from passing these readiness tests? Sounds similiar to how you can use any cni as long as it conformance to a base set of rules.

Reply:
Thanks for the clarification! Apologies about that, was doing too many things at once and had a short brain lapse when reading the service proxy portions.

Reply:
We have kpng as well, so agree on having an open statement about the proxy here